File: android/src/main/java/com/reactnativekalturaplayer/KalturaPlayerPackage.java
package com.reactnativekalturaplayer;

import androidx.annotation.NonNull;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class KalturaPlayerPackage implements ReactPackage {

   public KalturaPlayerPackage() {
   }

   KalturaPlayerViewManager kalturaPlayerViewManager;

   @NonNull
   @Override
   public List<ViewManager> createViewManagers(@NonNull ReactApplicationContext reactContext) {
      if (kalturaPlayerViewManager == null) {
         kalturaPlayerViewManager = new KalturaPlayerViewManager(reactContext);
      }
      return Collections.singletonList(kalturaPlayerViewManager);
   }

   @NonNull
   @Override
   public List<NativeModule> createNativeModules(@NonNull ReactApplicationContext reactContext) {
      if (kalturaPlayerViewManager == null) {
         kalturaPlayerViewManager = new KalturaPlayerViewManager(reactContext);
      }
      List<NativeModule> modules = new ArrayList<>();
      modules.add(new KalturaPlayerModule(reactContext, kalturaPlayerViewManager));
      return modules;
   }
}

File: android/src/main/java/com/reactnativekalturaplayer/KalturaPlayerPackage.kt
package com.reactnativekalturaplayer

import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext

class KalturaPlayerPackage: ReactPackage {

    private var kalturaPlayerViewManager: KalturaPlayerViewManager? = null

    override fun createNativeModules(reactContext: ReactApplicationContext): MutableList<NativeModule> {
        if (kalturaPlayerViewManager == null) {
            kalturaPlayerViewManager = KalturaPlayerViewManager(reactContext)
        }
        val modules: MutableList<NativeModule> = ArrayList()
        modules.add(KalturaPlayerModule(reactContext, kalturaPlayerViewManager))
        return modules
    }

    override fun createViewManagers(reactContext: ReactApplicationContext): List<KalturaPlayerViewManager?> {
        if (kalturaPlayerViewManager == null) {
            kalturaPlayerViewManager = KalturaPlayerViewManager(reactContext)
        }
        return listOf(kalturaPlayerViewManager)
    }
}
